File: test_struct.zig
const std = @import("std");

const JoltBodyDesc = extern struct {
    entity_id: u64,
    motion_type: u32,
    flags: u32,
    mass: f32,
    gravity_scale: f32,
    linear_damping: f32,
    max_linear_speed: f32,
    position: [3]f32,
    rotation: [4]f32,
    linear_velocity: [3]f32,
    box_half_extents: [3]f32,
    box_center: [3]f32,
    sphere_radius: f32,
    sphere_center: [3]f32,
    mesh_half_extents: [3]f32,
    mesh_center: [3]f32,
    layer_id: u32,
    layer_group: u32,
};

pub fn main() !void {
    std.debug.print("Size: {d}\n", .{@sizeOf(JoltBodyDesc)});
    std.debug.print("Offset of linear_velocity: {d}\n", .{@offsetOf(JoltBodyDesc, "linear_velocity")});
    std.debug.print("Offset of position: {d}\n", .{@offsetOf(JoltBodyDesc, "position")});
}

File: test_struct.cpp
#include <stdio.h>
#include <stddef.h>
#include <stdint.h>

struct GuavaJoltBodyDesc {
  uint64_t entity_id;
  uint32_t motion_type;
  uint32_t flags;
  float mass;
  float gravity_scale;
  float linear_damping;
  float max_linear_speed;
  float position[3];
  float rotation[4];
  float linear_velocity[3];
  float box_half_extents[3];
  float box_center[3];
  float sphere_radius;
  float sphere_center[3];
  float mesh_half_extents[3];
  float mesh_center[3];
  uint32_t layer_id;
  uint32_t layer_group;
};

int main() {
    printf("Size: %zu\n", sizeof(GuavaJoltBodyDesc));
    printf("Offset of linear_velocity: %zu\n", offsetof(GuavaJoltBodyDesc, linear_velocity));
    printf("Offset of position: %zu\n", offsetof(GuavaJoltBodyDesc, position));
    return 0;
}
